applications/asset_tracker/src/main.c

```diff
@@ -1447,24 +1460,24 @@ void cloud_event_handler(const struct cloud_backend *const backend,
 			break;
 		}
 
+#if defined(CONFIG_NRF_CLOUD_CELL_POS)
+		struct nrf_cloud_cell_pos_result result;
+
+		err = nrf_cloud_cell_pos_process(evt->data.msg.buf, &result);
+		if (err) {
```

Comment:
Only `break` if there is an error (negative) or cell pos data was processed.  Otherwise (err == 1), it is some other type of data that still needs to be processed.
```
if (err < 0) {			
        LOG_ERR("Failed to parse cell pos data, err: %d", err);
        break;		
} else if (err == 0) {
        LOG_INF("Cell-based location: %lf, %lf", result.lat, result.lon);		
        LOG_INF("Cell pos data processed");
        break;
}
```

Reply:
OK, will pass to AGPS lib for further processing after checking with original logic:
https://github.com/nrfconnect/sdk-nrf/blob/master/applications/asset_tracker/src/main.c#L1450

Reply:
@tokangas my suggested change above should prevent the assert you see in nrf_cloud_agps.c on line 649 when both are enabled.  
however, i think that assert should be removed and replaced with a negative return code `-EBADMSG`? Thoughts?

Reply:
In asset_tracker, if not cell_pos data then pass to agp lib for further handling.
In nrf_cloud, replace the assert with error return.

Reply:
I tried it with the latest commits and there's no assert anymore. Should it be possible to use cellular positioning and A-GPS at the same time? Both can be enabled at the same time in the configuration, but it seems that no A-GPS data request is sent to the cloud.